docs/messaging/usage/messaging-with-xmpp.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Messaging with XMPP
+description: To send and receive messages directly from devices with FCM requires you to create and deploy an XMPP server.
+---
+
+## Introduction
+
+This is a reference for using the Firebase Messaging service to send and receive messages directly between devices. Although methods (e.g. `sendMessage()` are provided for sending and receiving message, additional configuration is needed to ensure a working solution for direct messaging from devices.
+
+> A custom solution is needed to exchange messages directly between devices (including the message sender device)
+> please ensure a solution from this article has been configured to successfully send and receive messages from a device.
+
+The following describes how to set up a server to handle messages, including...
+
+- A custom XMPP server with XCS for receiving messages.
+- Firebase admin for sending messages.
+
+So what are we going to do?
+
+1. Send a message from a client device.
+2. Intercept using a custom XMPP server.
+3. Push the message to Firebase using firebase-admin
+4. Receive the message and acknowledge on other connected devices.
+
+## Why can't I send and receive messages?
```

Comment:
This section could cause confusion, it says non of these listeners will acknowledge a message or error - but they do? And also these methods work for non device to server communication so I think the distinction needs to be cleared up.

Alternatively just have this step show how to do `1. Send a message from a client device.` as per your bulleted points above, rather than all the extra bits.

Reply:
I see what you mean. It is is possible, however to use sendMessage() along with their listeners and for nothing to happen.  A few use cases have appeared where users are simply putting sendMessage and expecting to receive (all on the client)

Reply:
I prefer to leave section 2 in as directly addresses that use case that people are struggling with, what do you think?

Reply:
There have been issues as @dackers86 says, it's a valid point. If it's confusing I'd address language vs delete it then. I'd push it and directly solicit the folks that had the issue with a request for changes to make it clear (aka "merge and open for community PRs" ;-) ) but then I am assertively trying to engage others as an attempted style, with mixed success, though an encouraging amount of success